Makefile

```diff
@@ -1,58 +1,156 @@
-REGISTRY_SERVER_ADDRESS ?= ""
 
+# Default target when just running 'make'
+.DEFAULT_GOAL := all
+
+# Build targets
 TARGETS := karmada-dashboard-api \
-		   karmada-dashboard-web \
-
-
-# Build binary.
-#
-# Args:
-#   GOOS:   OS to build.
-#   GOARCH: Arch to build.
-#
-# Example:
-#   make
-#   make all
-#   make karmada-dashboard-api GOOS=linux
+		   karmada-dashboard-web
+
+# Docker image related variables
+REGISTRY				?= docker.io/karmada
+REGISTRY_USER_NAME  	?= 
+REGISTRY_PASSWORD   	?= 
+REGISTRY_SERVER_ADDRESS ?= 
+IMAGE_TARGET := $(addprefix image-, $(TARGETS))
+
+# Development server variables
+KARMADA_CTX ?= karmada-apiserver
+HOST_CTX ?= karmada-host
+API_PORT ?= 8000
+API_HOST ?= http://localhost:8000
+
+###################
+# Build Targets   #
+###################
+
+# Build all binaries (alias for build)
+.PHONY: all
+all: build
+
+# Build all binaries
+.PHONY: build
+build: $(TARGETS)
+
+# Build specific binary
 .PHONY: $(TARGETS)
 $(TARGETS):
 	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh $@
 
-.PHONY: all
-all: $(TARGETS)
-
-# Build image.
-#
-# Args:
-#   GOARCH:      Arch to build.
-#   OUTPUT_TYPE: Destination to save image(docker/registry).
-#
-# Example:
-#   make images
-#   make image-karmada-dashboard-api
-#   make image-karmada-dashboard-api GOARCH=arm64
-IMAGE_TARGET=$(addprefix image-, $(TARGETS))
+###################
+# Docker Images   #
+###################
+
+# Build all images
+.PHONY: images
+images: $(IMAGE_TARGET)
+
+# Build specific image
 .PHONY: $(IMAGE_TARGET)
 $(IMAGE_TARGET):
 	set -e;\
 	target=$$(echo $(subst image-,,$@));\
 	make $$target GOOS=linux;\
 	VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh $$target
 
+###################
+# UI Building     #
+###################
 
+.PHONY: bundle-ui-dashboard
 bundle-ui-dashboard:
 	cd ui && pnpm run dashboard:build
-bin-karmada-dashboard-web:
-	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh karmada-dashboard-web
-image-karmada-dashboard-web:
-	BUILD_PLATFORMS=linux/$(GOARCH) hack/build.sh karmada-dashboard-web
-	cp -R ui/apps/dashboard/dist _output/bin/linux/$(GOARCH)/dist
-	DOCKER_FILE="build-web.Dockerfile" VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh karmada-dashboard-web
 
+###################
+# Dependencies    #
+###################
+
+# Install Go dependencies
+.PHONY: install-deps
+install-deps:
+	go mod tidy
+	go mod verify
+
+# Install UI dependencies
+.PHONY: install-ui-deps
+install-ui-deps:
+	cd ui && pnpm install
+
+# Install all dependencies
+.PHONY: install
+install: install-deps install-ui-deps
+
+###################
+# Development     #
+###################
+
+# Run both API and Web server
+.PHONY: run
+run:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	@echo "Starting API server..."
+	@_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
```

Comment:
for my local case, I need `--skip-karmada-apiserver-tls-verify`, maybe add  an option to skip tls verify is a good idea?

Reply:
like following: 
![image](https://github.com/user-attachments/assets/085a5146-6d9e-4cfc-87ae-822b53f4c1cf)

Reply:
add an option for this. use

```bash
make run-api SKIP_TLS_VERIFY=true
```